images/ceph/set_ceph_debug_level.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Rook Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -x
+
+if [ -z "$1" ]; then
+  echo "no debug level passed choose between 0 and 20"
+  exit 1
+fi
+
+CEPH_DEBUG_LEVEL=$1
+CEPH_DEBUG_FLAG=(lockdep context crush mds mds_balancer mds_locker mds_log mds_log_expire mds_migrator buffer timer filer striper objecter rados rbd rbd_mirror rbd_replay journaler objectcacher client osd optracker objclass filestore journal ms mon monc paxos tp auth crypto finisher reserver heartbeatmap perfcounter rgw rgw_sync civetweb javaclient asok throttle refs compressor bluestore bluefs bdev kstore rocksdb leveldb memdb fuse mgr mgrc dpdk eventtrace)
```

Comment:
Maybe we could make a multiline assign rather than having the huge line? With arrays in bash, no `\` symbols are needed. Just

```sh
CEPH_DEBUG_FLAG=(
  lockdep
  context
  # ectc
)
```

Reply:
Would the user really want to enable the same log level on all the components? When they're done debugging, would they set them all back to level 1? I thought some components had different defaults.

Seems like they should only change the log level on the components they're interested in, instead of all components. In that case, maybe we just need to document how to enable the logging and no need for this script since they could copy commands from the docs into the toolbox.

Reply:
I think if we have a special value `default` (or just unset the env var?) log level, we can unset configs to let them go back to their builtin defaults. I think `ceph config rm <config>`.

Reply:
If we go into too many considerations like "which subsystem to enable" I believe this defeats the purpose of simplicity. The goal is here is for someone not familiar with Ceph to be asked to produce debug logs just to run a command and all daemons will log for all subsystems. IMO it's better to turn on everything so nothing is missed instead of going back and forth and adding more subsystems.

Reverting could simply take `default` as a special arg and then remove all the config entries.